userspace/libscap/scap_userlist.c

```diff
@@ -31,8 +31,8 @@ limitations under the License.
 //
 int32_t scap_create_userlist(scap_t* handle)
 {
-	uint32_t usercnt;
-	uint32_t grpcnt;
+	uint32_t usercnt, useridx;
+	uint32_t grpcnt, grpidx;
```

Comment:
Honestly, I'd rather we dropped the double walk over all users/groups.

We're saving some time allocating but spend potentially a lot more talking to nss (if e.g. the user database is in ldap, enumerating all users may be very slow).

I think we can just realloc() the table to double the size when we run out of space and then realloc() it back to the exact size once we finish enumerating users/groups.